sql/moz-fx-data-shared-prod/mozilla_vpn_derived/guardian_apple_events_v1/query.sql

```diff
@@ -0,0 +1,151 @@
+/*
+legacy subscriptions table is documented at:
+https://developer.apple.com/documentation/appstorereceipts/responsebody
+
+and the output for this table is documented at
+https://developer.apple.com/documentation/appstoreservernotifications/responsebodyv2decodedpayload
+*/
+WITH legacy_subscriptions AS (
+  SELECT
+    subscriptions.id,
+    users.fxa_uid AS user_id,
+    subscriptions.is_active,
+    subscriptions.updated_at,
+    mozfun.iap.parse_apple_receipt(subscriptions.provider_receipt_json) AS apple_receipt,
+  FROM
+    `moz-fx-data-shared-prod`.mozilla_vpn_external.subscriptions_v1 AS subscriptions
+  JOIN
+    `moz-fx-data-shared-prod`.mozilla_vpn_external.users_v1 AS users
+  ON
+    (subscriptions.user_id = users.id)
+  WHERE
+    subscriptions.provider = "APPLE"
+)
+SELECT
+  -- WARNING: mozdata.subscription_platform.apple_subscriptions and
+  -- mozdata.subscription_platform.nonprod_apple_subscriptions require field order of
+  -- moz-fx-data-shared-prod.mozilla_vpn_derived.guardian_apple_events_v1 to exactly match:
+  --   legacy_subscription_id,
+  --   event_timestamp,
+  --   mozfun.iap.parse_apple_event(`data`).*,
+  CAST(legacy_subscriptions.id AS STRING) AS legacy_subscription_id,
+  legacy_subscriptions.updated_at AS event_timestamp,
+  renewal_info.auto_renew_product_id,
+  renewal_info.auto_renew_status,
+  legacy_subscriptions.apple_receipt.receipt.bundle_id,
+  legacy_subscriptions.apple_receipt.environment,
+  renewal_info.expiration_intent,
+  receipt_info.expires_date,
+  CAST(NULL AS STRING) AS form_of_payment,
+  CAST(NULL AS TIMESTAMP) AS grace_period_expires_date,
+  receipt_info.in_app_ownership_type,
+  renewal_info.is_in_billing_retry,
+  CAST(NULL AS BOOL) AS is_upgraded,
+  CAST(NULL AS STRING) AS latest_notification_subtype,
+  CAST(NULL AS STRING) AS latest_notification_type,
+  receipt_info.offer_identifier,
+  receipt_info.offer_type,
+  receipt_info.original_purchase_date,
+  latest_original_transaction_id AS original_transaction_id,
+  receipt_info.product_id,
+  receipt_info.purchase_date,
+  receipt_info.revocation_date,
+  receipt_info.revocation_reason,
+  CASE -- https://developer.apple.com/documentation/appstoreserverapi/status
+  WHEN
+    TIMESTAMP_MILLIS(
+      legacy_subscriptions.apple_receipt.receipt.request_date_ms
+    ) < receipt_info.expires_date
+  THEN
+    1
```

Comment:
Unfortunately it looks like this is causing converted trials to end up as two subscription records, because the latest request date during the paid subscription is after the trial expiration date so the trial record ends up with `status = 2` and gets treated like a separate subscription period.

I'm not sure exactly the best way to do it, but maybe something like this?
```suggestion
  WHEN
    TIMESTAMP_MILLIS(
      legacy_subscriptions.apple_receipt.receipt.request_date_ms
    ) < receipt_info.expires_date
    OR (
      receipt_info.offer_type = 1
      AND COUNT(*) OVER (PARTITION BY latest_original_transaction_id) > 1
    )
  THEN
    1
```

Reply:
I solved this by adding a `source` field to receipt_info that is either `"receipt.in_app"` or `"latest_receipt_info"`, and using that here to do:

```
  WHEN
    receipt_info.source = "receipt.in_app"
  THEN
    1 -- subscription was active for the values in this receipt
```
and in `verified_at` to do:
```
  IF(
    receipt_info.source = "receipt.in_app",
    receipt_info.purchase_date,
    TIMESTAMP_MILLIS(legacy_subscriptions.apple_receipt.receipt.request_date_ms)
  ) AS verified_at,
```